From 52254c9f22ba7ef9d8af0c1f3d118fb6c56b0136 Mon Sep 17 00:00:00 2001
From: Rohith Sharma K S <rohithsharmaks@apache.org>
Date: Tue, 6 Sep 2016 14:09:45 +0530
Subject: [PATCH 1839/2848] YARN-5608. TestAMRMClient.setup() fails with
 ArrayOutOfBoundsException. Contributed by Daniel
 Templeton.

(cherry picked from commit f9557127b3660e138cab8d9535a589677809384a)
(cherry picked from commit de9f046598ac68b9726edee1d12f61cdd5f747db)

Change-Id: I000e7fea165023278cfcfcf79c4bd75c30b2a824
---
 .../yarn/client/api/impl/TestAMRMClient.java       |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestAMRMClient.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestAMRMClient.java
index 85dd6a1..e0137f7 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestAMRMClient.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestAMRMClient.java
@@ -135,8 +135,13 @@ public static void setup() throws Exception {
     yarnClient.start();
 
     // get node info
+    assertTrue("All node managers did not connect to the RM within the "
+        + "allotted 5-second timeout",
+        yarnCluster.waitForNodeManagersToConnect(5000L));
     nodeReports = yarnClient.getNodeReports(NodeState.RUNNING);
-    
+    assertEquals("Not all node managers were reported running",
+        nodeCount, nodeReports.size());
+
     priority = Priority.newInstance(1);
     priority2 = Priority.newInstance(2);
     capability = Resource.newInstance(1024, 1);
-- 
1.7.9.5

